### DISCUSSION TITLE
Asynchronous implementation

### DISCUSSION THREAD

**Discussion by captainyugi00** [SYSTRAN/faster-whisper#692] [Q&A] [answered]
Hello there, I was wondering if it's possible to use faster-whisper with async. I have not been trying by myself because I wanted to get a response from this before starting my implementation. Does anyone know if this is possible and/or is in the works?

**Comment by trungkienbkhn** [comment, marked as answer]
@captainyugi00 , hello. Yes, it is completely possible. Below is workflow in my test:
+ Firstly, init the work_queue to save transcribe request_id. For each transcribe request, we can create new request_id, put it to work_queue, then save request info (request_id, transcribe_option, input_file, status=queued, created_at, ...) to db (e.g. mongodb).
+ Next, create a Worker thread to process requests. This worker will read data from db then update result via request_id.
Status of request included: queued => running => completed (or failed)

If you want to check the request status, simply create an api to query into db via request_id

**Reply by captainyugi00** [reply]
Thank you, will definitely give it a try! Have you already been working on something similar?

**Reply by trungkienbkhn** [reply]
Yes, I maintain a python api server to transcribe/async_transcribe audio requests from clients.
